src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -752,35 +779,66 @@ protected bool TakeStartLine(SocketInput input)
                 if (chFound == '?')
                 {
                     begin = scan;
-                    if (scan.Seek(ref _vectorSpaces) != ' ')
+                    if (scan.Seek(ref _vectorSpaces) == -1)
                     {
-                        return false;
+                        return RequestLineStatus.TargetIncomplete;
                     }
                     queryString = begin.GetAsciiString(scan);
                 }
 
+                if (pathBegin.GetLength(scan) == 0)
```

Comment:
I don't like it either. I think I'll move this check to after we parse the request target, so I can just check against `string.Empty`.